code/src/cljs/sixsq/nuvla/ui/profile/views.cljs

```diff
@@ -1458,45 +1457,126 @@
+      (if (or @show-billing-contact
+              @show-payment-methods
+              @show-subscription
+              @show-coupon
+              @show-invoices)
         [ui/Grid {:stackable true
                   :centered  true}
          [ui/GridRow {:columns (grid-columns-dense @device)}
-          (for [s sub-sections]
-            ^{:key (random-uuid)}
-            [ui/GridColumn {:style {:padding-bottom "20px"}} [s]])]]
+          [ui/GridColumn {:style {:padding-bottom "20px"}}
+           (when @show-billing-contact [BillingContact])
+           (when @show-payment-methods [PaymentMethods])
+           (when @show-subscription [Vendor])
+           (when @show-coupon [Coupon])]
+          [ui/GridColumn {:style {:padding-bottom "20px"}}
+           (when @show-invoices [Invoices])]]]
+        [ui/Message {:info true}
+         (@tr [:no-subscription-information])]))))
+
+(defn EdgeCurrentConsumption
+  []
+  (let [tr               (subscribe [::i18n-subs/tr])
+        loading?         (subscribe [::subs/loading? :upcoming-invoice])
+        upcoming-invoice (subscribe [::subs/upcoming-invoice])
+        upcoming-lines   (subscribe [::subs/upcoming-invoice-lines])]
+    (fn []
+      [CurrentConsumptionView
+       {:upcoming-invoice @upcoming-invoice
+        :loading? @loading?
+        :upcoming-lines @upcoming-lines}])))
+
+(defn- GridRowWith2or1Cols [c]
+  (let [device (subscribe [::main-subs/device])]
+    [ui/GridRow {:columns (grid-columns-dense @device)}
+     c]))
+
+(defn GridColumPaddedBottom [c]
+  [ui/GridColumn {:style {:padding-bottom "20px"}} c])
+
+(defn AppsSubsAndConsumption
+  []
+  (let [tr                   (subscribe [::i18n-subs/tr])
+        apps-subs-and-consumptions (subscribe [::subs/apps-subscriptions-consumptions])
+        loading?         (subscribe [::subs/loading? :upcoming-invoice])]
+    (fn []
+      [:<>
+       (for [{:keys [upcoming-lines upcoming-invoice app-name subscription]}
+             @apps-subs-and-consumptions]
+         ^{:key app-name}
+         [ui/Card {:style {:width "100%"
+                           :padding "1rem"}}
+          [:h1 app-name]
+          [ui/Grid
+           [GridRowWith2or1Cols
+            [:<>
+             [GridColumPaddedBottom [SubscriptionCard subscription (@tr [:usage])]]
+             [GridColumPaddedBottom
+              [CurrentConsumptionView
+               {:label    app-name
+                :loading? @loading?
+                :upcoming-invoice upcoming-invoice
+                :upcoming-lines upcoming-lines}]]]]]])])))
+
+(defn Subscriptions
+  []
+  (let [tr                   (subscribe [::i18n-subs/tr])
+        show-subscription    (subscribe [::subs/show-subscription])
+        show-consumption     (subscribe [::subs/show-consumption])]
+    (fn []
+      (if-let [sub-sections (cond-> []
+
+                              ;; TODO: refactor back by removing `or true`
+                              (or true @show-subscription)
```

Comment:
this condition is always true on purpose?

Reply:
Fcuk no, that was for testing